mycroft/skills/common_query_skill.py

```diff
@@ -49,9 +51,30 @@ class CommonQuerySkill(MycroftSkill, ABC):
     This class works in conjunction with skill-query which collects
     answers from several skills presenting the best one available.
     """
+
     def __init__(self, name=None, bus=None):
         super().__init__(name, bus)
 
+        data_dir = Configuration.get().get("data_dir")
+        save_root_dir = self.root_dir
+        self.root_dir = data_dir + "/mycroft/"
+        noise_words_filename = self.find_resource("noise_words" + ".list", "res/text")
+        self.root_dir = save_root_dir
```

Comment:
Hi Ake, and thanks for the review. The removal of noise if you look in the actual loop uses a w = '  ' + noise_word + '  ' approach which is why variable phrase has a space added to the start and end, but there could always be a subtle bug. The problem with all the routines you reference for finding a resource is that they all assume the root dir is the current skill base but in this case I did not want to clone the noise words file for every skill so I added it once to the base class and use the root to override the base (as you can see i save and restore it). I do use the configured value for 'data_dir' but if you know of a routine for finding resource files in res/text/locale I am all ears. Based on our existing code i assumed mycroft/res/text/locale would always exist so the only issue is if 'data'_dir' is a valid location for resource files will always be located. I am flexible on this.

Reply:
The actual loop uses the word with the space when finding but without the space when replacing causing it to be able to remove too much. So if a " and " is found it will replace "and" with ""

I see I misunderstood the code a bit but it's still not quite right. The data dir /opt/mycroft/ in the normal case but you want the mycroft-core root folder. The function I refer to is the one in the mycroft.util module, and that refers to the MYCROFT_ROOT/res-folder and not the skill's folder.

Reply:
Thanks Ake, I'll update accordingly.

Reply:
So just so I am not missing something. This method still wants the language embedded in the file path. It does not do it for you - correct?

>>> from mycroft.util.file_utils import resolve_resource_file
>>> print(resolve_resource_file('text/en-us/noise_words.list'))
/opt/mycroft/mycroft/res/text/en-us/noise_words.list
>>> print(resolve_resource_file('noise_words.list'))
None
>>> print(resolve_resource_file('text/noise_words.list'))
None

Reply:
I had to manually add the language as shown above.